src/Microsoft.Framework.PackageManager/Restore/LockFileWriter.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
I think you should move the old lock file reader writer logic as a whole to the PackageManager and create a copy of the writer (unshared) in the runtime itself. The PackageManager needs to read the lock file as well but it can use the normal JSON.NET stuff.